actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb

```diff
@@ -14,15 +14,35 @@ module CollectionCaching # :nodoc:
       def cache_collection_render(instrumentation_payload)
         return yield unless @options[:cached]
 
+        # Result is a hash with the key represents the
```

Comment:
Is not better to document this in the method?

Reply:
If blame weren't an issue I would change some of the variables to indicate what they are i.e. `keyed_collection_hash`. 

The method return and behavior is actually somewhat easier to understand. 

This one method where I wrote all these docs is essentially where the "magic happens" and I had a hard time piecing together exactly how everything fit together. These are the slightly polished notes that I took when going through that process of understanding this code. Happy to make any changes you think that could aid in comprehension of how all the pieces fit together.

Reply:
I'm fine with it here too.